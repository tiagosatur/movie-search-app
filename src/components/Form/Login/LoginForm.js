import React from 'react';
import styled from 'styled-components';
import useForm from 'react-hook-form'
import { rem } from '../../../style';

import { Button, Input } from '../..';

const LoginForm = props => {
    const { onSubmit } = props
    const { register, handleSubmit, errors } = useForm();
    

    return(
        <div>
            <StyledForm onSubmit={handleSubmit(onSubmit)}>
                <Input 
                    name="username" 
                    placeholder='Username'
                    register={register({required: true})}
                    errors={errors}
                />
                <Input 
                    type='password' 
                    name='password' 
                    register={register({required: true})} 
                    placeholder='Password'
                    errors={errors}
                />
                <Button type='submit' solid>Login</Button>
                

            </StyledForm>
        </div>
    )
}

const StyledForm = styled.form`
   display: flex;
   flex-direction: column;
   width: 80%;
   margin: 0 auto;
   @media(min-width: ${rem(768)}) {
      width: 30%;
   }
`;

export default LoginForm;