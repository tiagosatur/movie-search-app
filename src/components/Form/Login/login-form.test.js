import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import LoginForm from './LoginForm';
import { spy } from 'sinon';


describe('LoginForm tests', () => {
  let wrapper;
  let mountedWrapper;
  const callback = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <LoginForm onSubmit={ callback } />
    );
  });
  
  it('should render the LoginForm', () => {
    expect(wrapper.find('.login-form').exists()).toBe(true);
  });

  it('should call LoginForm snapshot correctly', () => {
    const tree = renderer
        .create(wrapper)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });
})