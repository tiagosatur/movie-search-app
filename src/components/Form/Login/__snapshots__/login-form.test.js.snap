// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm tests should call LoginForm snapshot correctly 1`] = `
<div
  className="login-form"
>
  <form
    className="sc-fjdhpX kNBbKB"
    onSubmit={[Function]}
  >
    <div
      className="sc-htpNat bqYrTX"
    >
      <input
        className="input  sc-bxivhb izmEcM"
        id="login-input-username"
        name="username"
        placeholder="Username"
        type="text"
      />
    </div>
    <div
      className="sc-htpNat bqYrTX"
    >
      <input
        className="input  sc-bxivhb izmEcM"
        id="login-input-password"
        name="password"
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      className="button  sc-bdVaJa jllYbe"
      id="login-form-btn"
      title=""
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
